### PR TITLE
feat(active-release): add commit data to slack release notifications

### PR DESCRIPTION
**Pull request by barkbarkimashark**
Don't have a preview of the slack notification, but this formats the slack notification for a release notification to include commit data as part of a release. 

We have something like this in the email version of this notification - this PR adds the same thing for slack notifications.

### CODE REVIEW

File: src/sentry/integrations/slack/message_builder/issues.py [hunk 1]
```diff
@@ -438,15 +450,10 @@ def build(self) -> SlackBody:
             if self.last_release
             else "unknown"
         )
-        release_url = (
-            absolute_uri(
-                f"/organizations/{self.group.organization.slug}/releases/{quote(self.last_release.version)}/?project={self.group.project_id}"
```
[the remark this answers is not in the document]
Comment (scttcper) [reply src/sentry/integrations/slack/message_builder/issues.py:443]: it escapes characters to put into the url

File: src/sentry/notifications/notifications/rules.py [hunk 2]
```diff
@@ -250,7 +251,8 @@ def get_release_commits(self, release: Release) -> Sequence[CommitData]:
             for rc in release_commits
         ]
 
-    def release_url(self, release: Release) -> str:
+    @staticmethod
+    def release_url(release: Release) -> str:
         return str(
             absolute_uri(
                 f"/organizations/{release.organization.slug}/releases/{release.version}/?project={release.project_id}"
```
[the remark this answers is not in the document]
Comment (barkbarkimashark) [reply src/sentry/notifications/notifications/rules.py:258]: Right, version is any string basically. Could we escape the whole url when the slack builder retrieves the url from the context, or do we have to maintain two code paths for creating the url?

[the remark this answers is not in the document]
Comment (scttcper) [reply src/sentry/notifications/notifications/rules.py:258]: No that would break the slashes

[the remark this answers is not in the document]
Comment (barkbarkimashark) [reply src/sentry/notifications/notifications/rules.py:260]: Alright .. alright. I did a thing. Should be marginally better ..

**Review by scttcper — APPROVED** [review]

**Pull request merged**